internal/manifests/manifests.go

```diff
@@ -90,11 +88,50 @@ func (m *Manifests) CreateClusterManifestInternal(ctx context.Context, params op
 		return nil, err
 	}
 
+	if isCustomManifest {
+		err = m.createMetadataForManifest(ctx, params.ClusterID, path)
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	log.Infof("Done creating manifest %s for cluster %s", path, params.ClusterID.String())
 	manifest := models.Manifest{FileName: fileName, Folder: folder}
 	return &manifest, nil
 }
 
+func (m *Manifests) isUserManifest(ctx context.Context, clusterID string, folder string, fileName string) bool {
+	manifestMetadataPrefix := filepath.Join(clusterID, constants.ManifestMetadataFolder)
+	isCustomManifest, err := m.objectHandler.DoesObjectExist(ctx, filepath.Join(manifestMetadataPrefix, folder, fileName, constants.ManifestSourceUserSupplied))
+	if err != nil {
+		m.log.Warnf("Could not determine if manifest %s/%s is a %s manifest due to error %s, leaving this as nil", folder, fileName, constants.ManifestSourceUserSupplied, err.Error())
```

Comment:
As requested, I will propagate this error and make it visible by making it fatal.
This might lead to more actual failures when fetching manifests but will at least make those errors visible.